src/chains/ethereum/ethereum/src/blockchain.ts

```diff
@@ -1058,7 +1058,12 @@ export default class Blockchain extends Emittery<BlockchainTypedEvents> {
     if (this.#isPaused() || !this.#instamine) {
       return hash;
     } else {
-      if (this.#instamine && this.#options.miner.instamine === "eager") {
+      // if the transaction is not executable, we just have to return the hash
+      if (
+        this.#instamine &&
+        this.#options.miner.instamine === "eager" &&
+        isExecutable
```

Comment:
My main concern with this fix is that I don't think we currently have a way to return VM errors to transactions where we return the hash first. With eager mode we advertise that you can use the `vmErrorsOnRPCResponse` flag to get those VM errors, but that flag doesn't work with strict mode. This is pretty much reverting to strict mode functionality for future-nonce transactions; we aren't able to return VM errors.